src/profile_service.rs

```diff
@@ -104,7 +110,8 @@ fn test_custom_profile() {
     let profile_path = String::from(profile_dir.into_path()
                                     .to_str().unwrap());
 
-    let profile = ProfileService::new(Some(profile_path.to_owned()));
+    let profile = ProfileService::new(ProfilePath::Custom(profile_path
+                                                          .clone()));
```

Comment:
The reason we are doing the clone is because the ownership of this string gets passed to the profile service, and we need to do a comparison with this string in our assert statement here below. So doing a clone before assigning the string to a variable runs into the same ownership problem this clone is meant to address.